src/accelerate/commands/launch.py

```diff
@@ -140,7 +151,14 @@ def launch_command_parser(subparsers=None):
 
 
 def simple_launcher(args):
-    cmd = [sys.executable, args.training_script]
+    cmd = []
+    if args.no_python and args.module:
+        raise ValueError("--module and --no_python cannot be used together")
+    if not args.no_python:
+        cmd.append(sys.executable)
+        if args.module:
+            cmd.append("-m")
```

Comment:
Since both can't be true at the same time, the test should not be nested.
```suggestion
    elif args.module:
        cmd.append("-m")
```

Reply:
Good catch, thanks. I think after the un-indent, it still has to stay an `if` and not an `elif`, otherwise it won't get evaluated because the `if not args.no_python` above would always be evaluated `True` first.

```suggestion
    if args.module:
        cmd.append("-m")
```